jsx/Form.js

```diff
@@ -255,6 +255,256 @@ var SelectElement = React.createClass({
+        onKeyPress={this.handleKeyPress}
+      >
+        {emptyOptionHTML}
+        {Object.keys(options).map(function(option) {
+          return (
+            <option value={option} key={option}>{options[option]}</option>
+          );
+        })}
+      </select>;
+    } else {
+      input = <input
+        type="text"
+        name={this.props.name}
+        id={this.props.id}
+        className="form-control"
+        value={this.props.value || ""}
+        disabled={disabled}
+        onChange={this.handleChange}
+        onKeyPress={this.handleKeyPress}
+      />;
+    }
+
+    // iterate through added Tags items and render them
+    // with deletion button
+    let items = this.props.items.map(function(item) {
+      let itmTxt;
+      if (Object.keys(options).length > 0 && options[item] !== undefined) {
+        itmTxt = options[item];
+      } else {
+        itmTxt = item;
+      }
+      return (
+          <button
+            className="btn btn-info btn-inline"
+            type="button"
+            onClick={this.handleRemove}
+            data-item={item}
+          >
+            {itmTxt}
+            &nbsp;
+            <span
+              className="glyphicon glyphicon-remove"
+              data-item={item}
+            />
+          </button>
+      );
+    }, this);
+    return (
+      <div className={elementClass}>
+        <label className="col-sm-3 control-label" htmlFor={this.props.id}>
+          {this.props.label}
+          {requiredHTML}
+        </label>
+        <div className="col-sm-9">
+          {items}
+          {input}
+          {errorMessage}
+          <button
+            className="btn btn-success add"
+            id={this.props.id + 'Add'}
+            type="button"
+            onClick={this.handleAdd}
+            >
+            <span className="glyphicon glyphicon-plus"/>
+            {this.props.btnLabel}
+          </button>
+        </div>
+      </div>
+    );
+  }
+}
+
+TagsElement.propTypes = {
+  name: React.PropTypes.string.isRequired,
+  id: React.PropTypes.string.isRequired,
+  pendingValKey: React.PropTypes.string.isRequired,
+  options: React.PropTypes.object,
+  items: React.PropTypes.array,
+  label: React.PropTypes.string,
+  value: React.PropTypes.oneOfType([
+    React.PropTypes.string,
+    React.PropTypes.array
+  ]),
+  class: React.PropTypes.string,
+  multiple: React.PropTypes.bool,
+  required: React.PropTypes.bool,
+  disabled: React.PropTypes.bool,
+  emptyOption: React.PropTypes.bool,
+  errorMessage: React.PropTypes.string,
+  btnLabel: React.PropTypes.string,
+  onUserInput: React.PropTypes.func,
+  onUserAdd: React.PropTypes.func,
+  onUserRemove: React.PropTypes.func
+};
+
+TagsElement.defaultProps = {
+  name: '',
+  options: {},
+  items: [],
+  label: '',
+  value: undefined,
+  id: '',
+  class: '',
+  required: false,
+  disabled: false,
+  emptyOption: true,
+  hasError: false,
+  allowDupl: false,
+  useSearch: false,
```

Comment:
i don't believe useSearch is defined above.

also, is useSearch essentially an option that turns this FormElement into the Searchable dropdown element that was just merged last week?

Reply:
> also, is useSearch essentially an option that turns this FormElement into the Searchable dropdown element that was just merged last week?

good catch & yes